src/plugin.validation.dns.simply/Simply/SimplyDnsClient.cs

```diff
@@ -0,0 +1,100 @@
+            {
+                await DeleteRecordAsync(objectId, record.RecordId);
+            }
+
+            await CreateRecordAsync(objectId, new DnsRecord
+            {
+                Type = "TXT",
+                Name = domain,
+                Data = value,
+                Ttl = 3600,
+            });
+        }
+
+        public async Task DeleteRecordAsync(string identifier, string domain, string value)
+        {
+            var objectId = GetObjectId(identifier);
+            var records = await GetRecordsAsync(objectId);
+            var record = records.SingleOrDefault(x => x.Type == "TXT" && x.GetHostname(objectId) == domain && x.Data == value);
+            if (record is null)
+            {
+                _log.Warning($"The TXT record {domain} that should be deleted does not exist at Simply.");
+                return;
+            }
+
+            await DeleteRecordAsync(objectId, record.RecordId);
+        }
+
+        private async Task CreateRecordAsync(string objectId, DnsRecord record)
+        {
+            using var content = new StringContent(JsonSerializer.Serialize(record), Encoding.UTF8, "application/json");
+            using var response = await _httpClient.PostAsync(_baseUrl + $"/my/products/{WebUtility.UrlEncode(objectId)}/dns/records", content);
+            var responseBody = await response.Content.ReadAsStringAsync();
+            _log.Information("Simply responded with status {0}: {1}", (int)response.StatusCode, responseBody);
+            response.EnsureSuccessStatusCode();
+        }
+
+        private async Task DeleteRecordAsync(string objectId, int recordId)
+        {
+            using var response = await _httpClient.DeleteAsync(_baseUrl + $"/my/products/{WebUtility.UrlEncode(objectId)}/dns/records/{recordId}");
+            var responseBody = await response.Content.ReadAsStringAsync();
+            _log.Information("Simply responded with status {0}: {1}", (int)response.StatusCode, responseBody);
+            response.EnsureSuccessStatusCode();
+        }
+
+        private async Task<List<DnsRecord>> GetRecordsAsync(string objectId)
+        {
+            using var response = await _httpClient.GetAsync(_baseUrl + $"/my/products/{WebUtility.UrlEncode(objectId)}/dns/records");
+            _log.Information("Simply responded with status {0}", (int)response.StatusCode);
+            response.EnsureSuccessStatusCode();
+            await using var stream = await response.Content.ReadAsStreamAsync();
+            var records = await JsonSerializer.DeserializeAsync<DnsRecordList>(stream);
+            return records!.Records;
+        }
+
+
+        private static string EncodeBasicAuth(string account, string apiKey)
+        {
+            return Convert.ToBase64String(Encoding.UTF8.GetBytes(WebUtility.UrlEncode(account) + ":" + WebUtility.UrlEncode(apiKey)));
+        }
+
+        private static string GetObjectId(string identifier)
+        {
+            var parts = identifier.Split('.');

```

Comment:
This may not be correct in all cases, e.g. when using DNS substition (CNAMEs "forwarding" the challenge record to another domain) or when the provider supports multiple zones, one of the subdomain zones could be required. See some other plugins using the `FindBestMatch` helper to be more robust.